src/main/java/com/box/sdk/BoxSignRequest.java

```diff
@@ -0,0 +1,812 @@
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject responseJSON = JsonObject.readFrom(response.getJSON());
+        return new BoxSignRequest.Info(responseJSON);
+    }
+
+    /**
+     * Returns all the sign requests.
+     *
+     * @param api    the API connection to be used by the resource.
+     * @param fields the fields to retrieve.
+     * @return an iterable with all the sign requests.
+     */
+    public static Iterable<BoxSignRequest.Info> getAll(final BoxAPIConnection api, String... fields) {
+        return getAll(DEFAULT_LIMIT, api, fields);
+    }
+
+    /**
+     * Returns all the sign requests.
+     *
+     * @param limit  the limit of items per single response. The default value is 100.
+     * @param api    the API connection to be used by the resource.
+     * @param fields the fields to retrieve.
+     * @return an iterable with all the sign requests.
+     */
+    public static Iterable<BoxSignRequest.Info> getAll(int limit, final BoxAPIConnection api, String... fields) {
+        QueryStringBuilder queryString = new QueryStringBuilder();
+        if (fields.length > 0) {
+            queryString.appendParam("fields", fields);
+        }
+        URL url = SIGN_REQUESTS_URL_TEMPLATE.buildWithQuery(api.getBaseURL(), queryString.toString());
+        return new BoxResourceIterable<BoxSignRequest.Info>(api, url, limit) {
+
+            @Override
+            protected BoxSignRequest.Info factory(JsonObject jsonObject) {
+                BoxSignRequest signRequest = new BoxSignRequest(api, jsonObject.get("id").asString());
+                return signRequest.new Info(jsonObject);
+            }
+
+        };
+    }
+
+    /**
+     * Cancels a sign request if it has not yet been signed or declined.
+     * Any outstanding signers will no longer be able to sign the document.
+     *
+     * @return the created sign request's info.
+     */
+    public BoxSignRequest.Info cancel() {
+        URL url = SIGN_REQUEST_CANCEL_URL_TEMPLATE.buildAlphaWithQuery(getAPI().getBaseURL(), "", this.getID());
+        BoxJSONRequest request = new BoxJSONRequest(getAPI(), url, "POST");
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject responseJSON = JsonObject.readFrom(response.getJSON());
+        return new BoxSignRequest.Info(responseJSON);
+    }
+
+    /**
+     * Resends a sign request to all signers that have not signed yet. There is a 10 minute cooling-off period between
+     * sending an mail. If you make a resend call during the cooling-off period, BoxAPIException will be thrown.
+     *
+     */
+    public void resend() {
+        URL url = SIGN_REQUEST_RESEND_URL_TEMPLATE.buildAlphaWithQuery(getAPI().getBaseURL(), "", this.getID());
+        BoxJSONRequest request = new BoxJSONRequest(getAPI(), url, "POST");
+        BoxAPIResponse response = request.send();
+        response.disconnect();
```

Comment:
Why do we need to call `disconnect` here and not return any information from this method?